src/hooks/useCalculator.tsx

```diff
@@ -0,0 +1,80 @@
+import { useState } from 'react';
+import {
+    caculateResultInput,
+    isValidOperationInput,
+    isValidResultInput,
+} from '../utils';
+
+export const useCalculator = () => {
+    const [digitInput, setDigitInput] = useState<string>('');
+    const [operationInput, setOperationInput] = useState<string>('');
+    const [resultInput, setResultInput] = useState<string>('0');
+    const [operatorPrevNumber, setOperatorPrevNumber] = useState<string>('');
+
+
+    // 1.  들어오는 DigitInput에따라 resultInput 결과를 처리한다 .
+    const handleDigitInput = (e: React.MouseEvent<HTMLButtonElement>) => {
+        
+        // 1.1 숫자 유효성 체크
+        const currentDigitNumber = parseInt(
+            digitInput + e.currentTarget.value
+        ).toString(); //000입력되는거 방지
+
+        if (currentDigitNumber.length > 3) {
+            alert('숫자는 최대 3자리까지 입력가능합니다.');
+            return;
+        }
+
+        // 1.2 OperationInput 존재에 따라 ResultInput을 보여줄지 처리
+        if (!operationInput.length) { //operation이 존재 안하면 현재 숫자만 보여주면됨        
+            setResultInput(currentDigitNumber);
+            // Operator 이전 숫자를 저장하기 위해 사용
+            setOperatorPrevNumber(currentDigitNumber);
+        } else {
+            // 2. opeartion 존재할시 
+            //operation 이전에 있는 숫자 (operatorPrevNumber) + 현재 currentDigitNumber = resultINput
+            const currentResultInput =
+                operatorPrevNumber + operationInput + currentDigitNumber;
+            setResultInput(currentResultInput);
+        }
```

Comment:
구체적인 세부사항은 별도 함수로 분리해봐도 좋아요.
주석처럼 어떤 역할을 하는지 별도 함수로 네이밍하여 분리하고 상태를 업데이트만 해보는 식으로 작성해봅시다 :)
```suggestion
        if (!operationInput.length) { 
            setOperatorPrevNumber(currentDigitNumber);
        }
        
        const result = getResultValue(currentDigitNumber);
        setResultInput(currentResultInput);
```